### DISCUSSION TITLE
How can I make this kind of UI in pixi ways?

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7697] [Q&A]
![Kapture 2021-08-12 at 07 38 57](https://user-images.githubusercontent.com/87828192/129112771-5bfd23f2-5e1d-4fc3-9b4c-8bd6e3134eca.gif)

This picture is from music app I'm using  ('Ableton Live')
Currently I can only imagine using canvas context 2d. (document.creatElment('canvas') way.
But I don't want to create canvas element, it seems it costs a lot of memory.
What's your opinion?

**Comment by ivanpopelyshev** [comment]
ssame way but with Graphics. you can use graphics-smooth package if you want better AA on this curve.

As for interaction and everything - you have to do it yourself. PixiJS doesnt have UI , draggable points, e.t.c.

**Reply by humbleCoder1990** [reply]
![image](https://user-images.githubusercontent.com/87828192/129117203-90fbbd0e-d83f-4977-9490-e35ed1e8d33f.png)
I see no difference between SmoothGraphics (left), Graphics (right)
Should I tweak some settings?

```
const smoothGraphics = new SmoothGraphics()

smoothGraphics.beginFill(0xff0000)
smoothGraphics.drawRoundedRect(100, 100, 200, 200, 100)
smoothGraphics.endFill()

const graphics = new Graphics()

graphics.beginFill(0xff0000)
graphics.drawRoundedRect(310, 100, 200, 200, 100)
graphics.endFill()
```

**Reply by humbleCoder1990** [reply]
![image](https://user-images.githubusercontent.com/87828192/129117720-a1887660-c2d6-4514-8b6f-0c01ddf67edf.png)

Wow, on Circle, the difference is huge!
I missed 3rd parameter (true)
Why do i need 3rd paramter? Can you just make it as 'built-in'?

**Reply by humbleCoder1990** [reply]
![image](https://user-images.githubusercontent.com/87828192/129117881-9d91b54d-1091-41bf-868c-865c4a65d58a.png)
Ivan, it's amazing.
Please add support for 'RoundedRect'.

RoundedRect is most important shape in GUI...these days..

**Reply by humbleCoder1990** [reply]
![image](https://user-images.githubusercontent.com/87828192/129118911-eed65a0b-44fd-4c8a-ad77-f98a01b33f80.png)
It's king of stroke :)

**Reply by humbleCoder1990** [reply]
![image](https://user-images.githubusercontent.com/87828192/129120151-f4681463-bc65-41cc-b9bc-f22eaf6a1515.png)

I could create rounded fill by layering SmoothGraphics + Sprite.

```
const rectX = 50
const rectY = 50
const rectW = 200
const rectH = 200
const borderWidth = 40
const borderRadius = 50
const color = 0xff0000

const border = new SmoothGraphics()

border.lineStyle(borderWidth, color)
border.drawRoundedRect(rectX + (borderWidth / 2), rectY + (borderWidth / 2), rectW - (borderWidth), rectH - (borderWidth), borderRadius)

const fill = new Sprite(Texture.WHITE)

fill.x = rectX + (borderWidth / 2)
fill.y = rectY + (borderWidth / 2)
fill.width = rectW - (borderWidth)
fill.height = rectH - (borderWidth)
fill.tint = color
```

**Reply by bigtimebuddy** [reply]
Fills are not supported yet with SmoothGraphics only stoke.

**Reply by ivanpopelyshev** [reply]
Not all fills are supported yet, RoundedRect isnt :(
